common/packets/event.h

```diff
@@ -25,11 +25,45 @@
 
 namespace RemoteCL
 {
+template<PacketType Type>
+struct EventCallback final : public Packet
```

Comment:
You don't seem to be using this Packet template anywhere else. Does it need to be a template?